include/El/optimization/bfgs.hpp

```diff
@@ -0,0 +1,226 @@
+/*
+   Copyright (c) 2009-2016, Ryan H. Lewis
+   All rights reserved.
+
+   This file is part of Elemental and is under the BSD 2-Clause License, 
+   which can be found in the LICENSE file in the root directory, or at 
+   http://opensource.org/licenses/BSD-2-Clause
+*/
+#ifndef EL_OPTIMIZATION_BFGS_HPP
+#define EL_OPTIMIZATION_BFGS_HPP
+#include <utility>
+#include <El.hpp>
+
+namespace El {
+
+template< typename T, typename Function, typename Gradient>
+T Zoom( const Function& f, const Gradient& gradient, T f0, const DistMatrix<T>& x0, const DistMatrix<T>& p,
```

Comment:
These parameters match the notation in nocedal and wright. I'd prefer to leave them and mention that explicitly. Is that acceptable?